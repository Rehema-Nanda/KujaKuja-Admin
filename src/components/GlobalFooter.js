import React from "react";
import "./GlobalFooter.css";

const footerLogo = require("../img/footer-logo.svg");

const GlobalFooter = () => {
    return (
        <section className="footer">
            <div className="footer-details">
                <div className="footer-logo">
                    <img alt="Logo" src={footerLogo} />
                </div>
                <div className="footer-social">
                    <div
                        className="social-instagram"
                        onClick={() => {
                            window.open("https://www.instagram.com/kujakujaglobal/", "_blank");
                        }}
                    />
                    <div
                        className="social-fb"
                        onClick={() => {
                            window.open("https://www.facebook.com/KujaKujaGlobal", "_blank");
                        }}
                    />
                    <div
                        className="social-twitter"
                        onClick={() => {
                            window.open("https://twitter.com/kujakujaglobal", "_blank");
                        }}
                    />
                    <div
                        className="social-medium"
                        onClick={() => {
                            window.open("https://medium.com/@KujaKujaGlobal", "_blank");
                        }}
                    />
                </div>
            </div>
        </section>
    );
};

export default GlobalFooter;
