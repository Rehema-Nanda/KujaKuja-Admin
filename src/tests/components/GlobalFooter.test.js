import React from "react";
import { shallow } from "enzyme";
import GlobalFooter from "../../components/GlobalFooter";

window.open = jest.fn();

describe("<GlobalFooter />", () => {
    const wrapper = shallow(<GlobalFooter />);

    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("redirects to social media links when icons are clicked", () => {
        wrapper.find("div.social-instagram").simulate("click");
        wrapper.find("div.social-fb").simulate("click");
        wrapper.find("div.social-twitter").simulate("click");
        wrapper.find("div.social-medium").simulate("click");
        expect(window.open).toHaveBeenCalledTimes(4);
    });
});
