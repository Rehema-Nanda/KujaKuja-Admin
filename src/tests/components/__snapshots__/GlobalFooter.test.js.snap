// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalFooter /> renders correctly 1`] = `
"<section className=\\"footer\\">
  <div className=\\"footer-details\\">
    <div className=\\"footer-logo\\">
      <img alt=\\"Logo\\" src={{...}} />
    </div>
    <div className=\\"footer-social\\">
      <div className=\\"social-instagram\\" onClick={[Function: onClick]} />
      <div className=\\"social-fb\\" onClick={[Function: onClick]} />
      <div className=\\"social-twitter\\" onClick={[Function: onClick]} />
      <div className=\\"social-medium\\" onClick={[Function: onClick]} />
    </div>
  </div>
</section>"
`;
